.github/workflows/check_hash.yml

```diff
@@ -0,0 +1,32 @@
+name: Compare File Hash Weekly
+
+on:
+  schedule:
+    - cron: '0 0 * * 0' # Runs weekly on Sunday at midnight UTC
+
+jobs:
+  compare-hash:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Calculate Local File Hash
+        id: local-hash
+        run: echo "LOCAL_HASH=$(sha256sum pysquirrel/data/NUTS2021-NUTS2024.xlsx | awk '{print $1}')" >> $GITHUB_ENV
+
+      - name: Download File
+        run: curl -o most-recent-version.xlsx "https://ec.europa.eu/eurostat/documents/345175/629341/NUTS2021-NUTS2024.xlsx"
```

Comment:
Does this actually always fetch the most recent version? Does the URL, either in the file name or the preceding address change? If so, this workflow would of course just always download the same file regardless of whether or not it's the latest NUTS file.

Reply:
As far as I can tell, this is the URL that always points to the most recent version. This URL redirects to the full URL with the download (the full URL has a hash on the end that changes with each version, [that's why the previous pysquirrel version broke, as a new file was being downloaded and the pooch hash check failed](https://github.com/iiasa/pysquirrel/issues/38)).

Reply:
Ah ok that's good then.